### ISSUE TITLE
How do I know if a user is logged in?

### ISSUE THREAD

**Issue by jeff-h** [RESTful-Drupal/restful#670]
How can I tell if the client is currently logged in? I need to present a dialog box to the user requesting that they log in, but if they are already logged in (i.e. their cookie is still valid) I don't want to show the box.

Also, how do I know which UID this user is, so I can fetch their full user info?

Prefer using the cookie-based authentication, if that has an impact.

**Comment by e0ipso** [comment]
I am not sure that you can do that out of the box.

In RESTful we don't log users in. Instead, we authenticate a request and perform the needed access checks against the resolved user.

For your particular use case, I'd extend the `users` resource to respond to the special ID `'me'`. That way you could do `curl https://example.org/api/users/me?fields=uid` and get the uid of the authenticated user.

**Comment by e0ipso** [comment]
Please feel free to reopen the issue.

**Comment by jeff-h** [comment]
Thanks for that. I can't actually reopen an issue — I think only Owners can do that?

Is it possible to extend an existing class without creating a new class, or do I have to create a new version of the resource? I'd prefer to extend, since this isn't a resource I have other plans for, but I'm not certain what's best practice.

The way I went just to get this running was to create a `api/v1.1/users`. I documented this here: 

https://github.com/RESTful-Drupal/restful/wiki/Obtaining-the-current-user
